catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/transformer/CswActionTransformerProvider.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.ogc.csw.catalog.endpoint.transformer;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.Nullable;
+import javax.xml.namespace.QName;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Manages a reference list of {@link CswActionTransformer}'s by mapping them to the QName's they
+ * apply to.
+ */
+public class CswActionTransformerProvider {
+  private Map<QName, CswActionTransformer> transformerMap = new ConcurrentHashMap<>();
+
+  private final Map<String, QName> typeNameQNameMap = new HashMap<>();
+
+  public synchronized void bind(CswActionTransformer cswActionTransformer) {
+    if (cswActionTransformer == null) {
+      return;
+    }
+
+    List<QName> namespaces = cswActionTransformer.getQNames();
+    for (QName namespace : namespaces) {
+      transformerMap.put(namespace, cswActionTransformer);
+      List<String> typeNames = cswActionTransformer.getTypeNames();
+
+      for (String typeName : typeNames) {
+        typeNameQNameMap.put(typeName, namespace);
+      }
+    }
+  }
+
+  public synchronized void unbind(CswActionTransformer cswActionTransformer) {
+    if (cswActionTransformer == null) {
+      return;
+    }
+
+    List<QName> namespaces = cswActionTransformer.getQNames();
+    for (QName namespace : namespaces) {
+      transformerMap.remove(namespace);
+      List<String> typeNames = cswActionTransformer.getTypeNames();
+      for (String typeName : typeNames) {
```

Comment:
Is it possible for typeNames to be null?

Reply:
I don't think so, since I already added a check to prevent `bind()` from adding `CswActionTransformer`s to the `transformerMap` if `typeNames` is null, so if a `CswActionTransformer` is being unbound, then it must have at least one typename.